### DISCUSSION TITLE
Redirect subdomains to different services, using same cert

### DISCUSSION THREAD

**Discussion by mottech20** [nginx-proxy/nginx-proxy#2224] [Q&A]
Currently I have nginx-proxy listening on my domain (mydomain.com and *.mydomain.com) but I want to add another service.
Let's say I have two different services with their own docker-compile file. Lets call them ServiceA and ServiceB.
How can I redirect serv-a.mydomain.com to ServiceA and serv-b.mydomain.com to ServiceB? Preferably using the same LE certificate.

Do I need to declare VIRTUAL_HOST and LETSENCRYPT_HOST in each of the services' docker-compose? In that case would one not override the other?
Also, should they use the same nginx-proxy network? How about port mapping?

Help appreciated.

PS: I did see https://github.com/nginx-proxy/nginx-proxy/discussions/2169 but that seems to be using only one service.

**Comment by SchoNie** [comment]
Yes, looks like you are describing the exact use case of nginx-proxy. I think you should not focus on using the same cert. The idea of LE is to have a separate (free) cert for each (sub)domain you want. And the acme-companion all handles this for you.
The LE cert is not a wildcard so using the same cert will not work on subdomains.
 
On serv-a.mydomain.com you set `VIRTUAL_HOST=serv-a.mydomain.com`  `LETSENCRYPT_HOST=serv-a.mydomain.com` 
and on the second service serv-b.mydomain.com you set `VIRTUAL_HOST=serv-b.mydomain.com` `LETSENCRYPT_HOST=serv-b.mydomain.com` 
Make sure they are in the same network as nginx-proxy. (if using differenty compose files specify [custom networks](https://docs.docker.com/compose/networking/#specify-custom-networks))
Spin the docker compose files up and nginx-proxy will detect them.

Port mappings I do not understand. There are only port mappings needed to the main nginx-proxy container. The proxy connections to the services happen internally in the docker network. nginx-proxy will use the exposed (dockerfile) port from the service. But you can also set `VIRTUAL_PORT` 
If the services A + B use the same port, that does not conflict as normally you do not map the ports to the host.

If you need more help, posting the code you have so far could help!
